packages/strapi-plugin-content-type-builder/admin/src/components/CustomCheckbox/index.js

```diff
@@ -66,7 +66,7 @@ class CustomCheckbox extends React.Component {
             errors={errors}
             name={name}
             onChange={this.handleInputNumberChange}
-            value={value || ''}
+            value={value && Number(value)}
```

Comment:
Here, the props `value={value}` is here enough.

But to work, the state should be 
`state = {
    isChecked:
      this.props.value !== null &&
      this.props.value !== undefined &&
      !isNaN(this.props.value),
  };`
at line 17.

Can you replace both?